plv8.cc

```diff
@@ -260,96 +297,236 @@ CreateIsolate(plv8_context *context) {
+								gettext_noop("V8 execution timeout."),
+								gettext_noop("The default value is 300 seconds.  "
+											 "This allows you to override the default execution timeout."),
+								&plv8_execution_timeout,
+								300, 1, 65536,
+								PGC_SUSET, 0,
+#if PG_VERSION_NUM >= 90100
+								NULL,
+#endif
+								NULL,
+								NULL);
+	}
+#undef EXECUTION_TIMEOUT_VAR
+#endif
+
+#define MEMORY_LIMIT_VAR "plv8.memory_limit"
+	guc_value = plv8_find_option(MEMORY_LIMIT_VAR);
+	if (guc_value != NULL) {
+		plv8_memory_limit = plv8_int_option(guc_value);
+	} else {
+		DefineCustomIntVariable(MEMORY_LIMIT_VAR,
+								gettext_noop("Per-isolate memory limit in MBytes"),
+								gettext_noop("The default value is 256 MB"),
+								(int *) &plv8_memory_limit,
+								256, 256, 3096, // hardcoded v8 limits for isolates
+								PGC_SUSET, 0,
 #if PG_VERSION_NUM >= 90100
-							NULL,
+								NULL,
 #endif
-							NULL,
-							NULL);
+								NULL,
+								NULL);
+	}
+#undef MEMORY_LIMIT_VAR
+
+#define USER_CONTEXT_VAR "plv8.context"
+	guc_value = plv8_find_option(USER_CONTEXT_VAR);
+	if (guc_value != NULL) {
+		plv8_user_context = plv8_string_option(guc_value);
+	} else {
+		DefineCustomStringVariable(USER_CONTEXT_VAR,
+								   gettext_noop("User-defined context id"),
+								   gettext_noop("Can be used to set up different execution contexts"),
+								   &plv8_user_context,
+								   nullptr,
+								   PGC_USERSET, 0,
+#if PG_VERSION_NUM >= 90100
+								   &user_context_check_hook,
+#endif
+								   nullptr,
+								   nullptr);
+	}
+#undef USER_CONTEXT_VAR
+
+#define USER_CONTEXT_SIZE_VAR "plv8.context_cache_size"
+	guc_value = plv8_find_option(USER_CONTEXT_SIZE_VAR);
+	if (guc_value != NULL) {
+		plv8_context_cache_size = plv8_int_option(guc_value);
+	} else {
+		DefineCustomIntVariable(USER_CONTEXT_SIZE_VAR,
+								gettext_noop("Number of live contexts in user context cache"),
+								gettext_noop("The default is 8, which means 10 contexts overall (+ default + compile)."
+											 "The cache will evict and destroy contexts over this number (using LRU)"),
+								(int *)&plv8_context_cache_size,
+								8, 0, 64,
+								PGC_SUSET, 0,
+#if PG_VERSION_NUM >= 90100
+								NULL,
 #endif
+								NULL,
+								NULL);
+	}
+#undef USER_CONTEXT_SIZE_VAR
 
-	DefineCustomIntVariable("plv8.memory_limit",
-							gettext_noop("Per-isolate memory limit in MBytes"),
-							gettext_noop("The default value is 256 MB"),
-							(int*)&plv8_memory_limit,
-							256, 256, 3096, // hardcoded v8 limits for isolates
-							PGC_SUSET, 0,
+#define MAX_EVAL_SIZE_VAR "plv8.max_eval_size"
+	guc_value = plv8_find_option(MAX_EVAL_SIZE_VAR);
+	if (guc_value != NULL) {
+		plv8_max_eval_size = plv8_int_option(guc_value);
+	} else {
+		DefineCustomIntVariable(MAX_EVAL_SIZE_VAR,
+								gettext_noop("Max size in bytes for code to eval()"),
+								gettext_noop("The default is 2MB, setting to 0 disables eval() completely!"
```

Comment:
Probably need to change it to -1 by default for upstream to maintain existing behavior...

Reply:
yes please, if you could in the PR.

Reply:
Done